pupil-spa/src/app/check/check.component.ts

```diff
@@ -234,4 +261,81 @@ export class CheckComponent implements OnInit {
     this.allowedStates.push('complete');
     // console.log('check.component: initStates(): states set to: ', this.allowedStates);
   }
+
+  /**
+   * Handle a page refresh
+   */
+  refreshDetected() {
+    const stateDesc = this.getStateDescription();
+    // console.log(`Refresh detected during state ${this.state} ${stateDesc}`);
+    this.auditService.addEntry(new RefreshDetected());
+
+    // Lets say that handling reloads during the check should always show the current screen
+    // in which case handling the reload whilst a question was being shown is a special case.
+    if (CheckComponent.questionRe.test(stateDesc)) {
+      // the page was reloaded when a question was shown
+      // Store the answer as the empty string (as there was no input)
+      // we need to initialise the current question, with the one from the current state
+      const matches = CheckComponent.questionRe.exec(stateDesc);
+      const questionNum = parseInt(matches[ 1 ], 10);
```

Comment:
isn't radix (second arg) defaulting to 10 anyway?

Reply:
The recommendation is always use it, so I do :)
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt